docs(README): update block and allow terminology (#1512)
jquense/yup: README.md

#### `mixed.oneOf(arrayOfValues: Array<any>, message?: string | function): Schema` Alias: `equals`

> Allowlist a set of values. Values added are automatically removed from any blocklist if they are in it.

Reviewer: ```suggestion
Allow a specific set of values. Values added are automatically removed from any blocklist if they are in it.
```

#### `mixed.notOneOf(arrayOfValues: Array<any>, message?: string | function)`

> Blocklist a set of values. Values added are automatically removed from any allowlist if they are in it.

Reviewer: ```suggestion
Disallow values from a set of values. Values added are removed from `oneOf` values if present.
```
